### PR TITLE
[GSK-2153] removing mlruns artifacts file produced by pytest

### PR DESCRIPTION
**Pull request by rabah-khalek**
very marginal improvement

this is what I get before:
tests/integrations/test_mlflow.py::test_fast[drug_classification_data-drug_classification_model],166.398,593.125
tests/integrations/test_mlflow.py::test_fast[diabetes_dataset_with_target-linear_regression_diabetes],14.398,607.559
tests/integrations/test_mlflow.py::test_fast[hotel_text_data-hotel_text_model],233.000,840.559

[3:25](https://giskardai.slack.com/archives/D05RXHQQMSB/p1699971911245999)
this is after a fix:
tests/integrations/test_mlflow.py::test_fast[drug_classification_data-drug_classification_model],177.035,600.938
tests/integrations/test_mlflow.py::test_fast[diabetes_dataset_with_target-linear_regression_diabetes],14.121,615.059
tests/integrations/test_mlflow.py::test_fast[hotel_text_data-hotel_text_model],227.391,842.480

[3:25](https://giskardai.slack.com/archives/D05RXHQQMSB/p1699971926018669)
basically the file that was used to be produced weighs
860K    mlruns

### CODE REVIEW

**Comment by mattbit** [comment]
It seems our hypothesis was wrong @rabah-khalek @Hartorn:

```
mlflow.exceptions.MlflowException: Could not find a registered artifact repository for: c:\Users\RUNNER~1\AppData\Local\Temp\tmpeieeqloy/da1d290ebe524d5e812f8fb5d02697b9/artifacts
```

:(

File: tests/integrations/test_mlflow.py [hunk 1]
```diff
@@ -10,20 +12,21 @@
...
-        model=model_info.model_uri,
-        model_type=mlflow_model_types[model.meta.model_type],
-        data=dataset.df,
-        targets=dataset.target,
-        evaluators="giskard",
-        evaluator_config=evaluator_config,
-    )
-    mlflow.end_run()
+    with TemporaryDirectory() as f:
+        mlflow.set_tracking_uri(Path("file://" + f))
```
Comment (mattbit) [inline tests/integrations/test_mlflow.py:18]: This is not a `Path`, it looks like a URI. Use `file:///path/to/the/dir`.

**Review by mattbit — CHANGES_REQUESTED** [review]
Some problems with the paths.

File: tests/integrations/test_mlflow.py (hunk 1 above)
Comment (mattbit) [reply tests/integrations/test_mlflow.py:18]: If you want to pass a `Path` instance, just do

```python
mlflow.set_tracking_uri(Path(f))
```

Comment (rabah-khalek) [reply tests/integrations/test_mlflow.py]: you're right, thanks!

**Comment by rabah-khalek** [comment]
it's working now

**Review by mattbit — APPROVED** [review]
